jujugui/static/gui/src/app/assets/javascripts/jujulib/test-jujulib-environment.js

```diff
@@ -186,92 +233,128 @@ describe('jujulib environment manager', function() {
     var currentUser = {user: 'test'};
     var bakery = {
       sendGetRequest: function(path, success, failure, redirect) {
-        assert.equal(path, 'http://example.com/v1/whoami');
+        assert.equal(path, 'http://example.com/v2/whoami');
         // Make sure that we have disabled redirect on 401
         assert.strictEqual(redirect, false);
         var xhr = _makeXHRRequest(currentUser);
         success(xhr);
       },
     };
 
-    env = new window.jujulib.environment('http://example.com/', 'v1', bakery);
+    env = new window.jujulib.jem('http://example.com/', bakery);
     env.whoami(
       function(error, data) {
-        if (error) {
-          assert.fail('callback should be successful');
-        } else {
-          assert.deepEqual(data, currentUser);
-        }
+        assert.strictEqual(error, null);
+        assert.deepEqual(data, currentUser);
         done();
       }
     );
   });
 
-  it('lists the current user\'s templates', function(done) {
-    var response = {templates: [{path: 'rose/template'}]};
+  it('lists templates', function(done) {
```

Comment:
"the current user's templates" is actually more correct, isn't it? It doesn't list all templates. Not really important, just noting.